gufe/settings/models.py

```diff
@@ -0,0 +1,174 @@
+    potential: Literal["Lennard-Jones-12-6"]
+    scale: vdWScale
+    long_range_dispersion: Literal["isotropic"]
+    cutoff: float  # angstrom
+    switch_width: float  # angstrom
+    method: Literal["cutoff"]
+
+
+class ElectrostaticSettings(SettingsBaseModel):
+    """
+    Settings for electrostatics
+    """
+
+    # Tricky since this allows functions https://openforcefield.github.io/standards/standards/smirnoff/#electrostatics
+    periodic_potential: Union[Literal["Ewald3D-ConductingBoundary"], str]
+    nonperiodic_potential: Union[Literal["Coulomb"], str]
+    exception_potential: Union[Literal["Coulomb"], str]
+    scale: ElectrostaticScale
+    cutoff: Union[float, None]
+    switch_width: Union[float, None]
+    solvent_dielectric: Union[float, None]
+
+
+class BondSettings(SettingsBaseModel):
+    """
+    Settings for harmonic bonds
+    """
+
+    # U(r) = (k/2)*(r-length)^2
+    potential: Literal["harmonic"]
+    # Might support other methods?
+    fractional_bondorder_method: Literal["AM1-Wiberg"]
+    fractional_bondorder_interpolation: Literal["linear"]
+
+
+class AngleSettings(SettingsBaseModel):
+    """
+    Settings for harmonic angles
+    """
+
+    # U(r) = (k/2)*(theta-angle)^2
+    potential: Literal["harmonic"]
+
+
+class ProperTorsionSettings(SettingsBaseModel):
+    """
+    Settings for proper torsions
+    """
+
+    # U = \sum_{i=1}^N k_i * (1 + cos(periodicity_i * phi - phase_i))
+    potential: Literal["k*(1+cos(periodicity*theta-phase))"]
+    default_idivf: Union[Literal["auto"], int] = "auto"
+    # Might support other methods?
+    fractional_bondorder_method: Literal["AM1-Wiberg"]
+    fractional_bondorder_interpolation: Literal["linear"]
+
+
+class ImproperTorsionSettings(SettingsBaseModel):
+    """
+    Settings for improper torsions
+    """
+
+    # U = \sum_{i=1}^N k_i * (1 + cos(periodicity_i * phi - phase_i))
+    potential: Literal["k*(1+cos(periodicity*theta-phase))"]
+    default_idivf: Literal["auto"]
+
+
+class GBSASettings(SettingsBaseModel):
+    """ "
+    Settings for Generalized-Born surface area (GBSA) implicit solvent parameters
+    """
+
+    gb_model: Literal["HCT", "OBC1", "OBC2"]
+    solvent_dielectric: float
+    solute_dielectric: float
+
+
+class ForcefieldSettings(SettingsBaseModel):
+    """
+    Settings for the forcefield
+    """
+
+    # Metadata
+    date: Union[date, None]
+    author: Union[str, None]
+
+    # These should also allow None
+    vdW: VdWSettings
+    electrostatics: ElectrostaticSettings
+    bonds: BondSettings
+    angles: AngleSettings
+    proper_torsions: ProperTorsionSettings
+    improper_torsions: ImproperTorsionSettings
+    gbsa: GBSASettings
+
+    # Misc
+    aromaticity_model: Literal["OEAroModel_MDL"]
+
+
+class ThermoSettings(SettingsBaseModel):
+    """
+    Settings for thermodynamic parameters
+    """
+
+    temperature: PositiveFloat  # Θ
+    pressure: PositiveFloat  # M L**−1 T**−2
```

Comment:
Maybe this is what you're making with Matt in openff-pydantic, but ideally these'll be in temperature/pressure units (which validate that they're positive)

Reply:
Also temperature and pressure are also optional (NVE/NVT)

Reply:
might also depend on the integrator of choice (MMC, MD, Langevin are all different here)

Reply:
I agree with @richardjgowers that this needs to be at the protocol level.

Off the top of my head I can't think of a single case of a microcanonical calculation being used as part of a free energy calculation, but someone out there might be doing some real cool with stat mech that my pharmacist brain can't conceptualize.

I'd even venture further than what Richard is saying and propose that maybe ensemble checking shouldn't even happen here, even if someone says they'll do something silly like NPE, then the protocol should define the range of ensembles it'll take and everything else just gets told "no"?

W.r.t. integrators - yeah we'll need to accept the full range of integrators in the relevant setting for it, for non-alchemical perturbations you might want to rely on someone like gromac's `md` integrator to speed things up (e.g. in an MMPBSA style calculation where you just need a vanilla MD sim).

Reply:
If we are not including the integrator or ensemble, the best we can do is validate the inputs are the correct dimension + sign (ie no negative temperature).